guide_drafts/model-derives.md

```diff
@@ -0,0 +1,769 @@
+Model Derives in Depth
+===============
+
+Diesel lets you write safe queries by using traits that you can implement on your models. Writing these implementations by hand can be very laborious, so in addition to the `diesel` crate, the `diesel_codegen` crate offers custom derives. In this section, we will cover each trait in detail in terms of its use cases, usage considerations, as well compare hand written implementations to its derive. To make use of these derives, `diesel_codegen` must be imported into the root of your project.
+
+Throughout this guide, we will be looking at examples for each trait and how they interact together. Some of the code will be implementing basic CRUD operations, however the details of those operations will be not be covered beyond their relevance to the demonstrated trait.
+
+- [Queryable](#queryable)
+- [Insertable](#insertable)
+- [Identifiable](#identifiable)
+- [AsChangeset](#aschangeset)
+- [Associations](#associations)
+
+## <a name="queryable"></a>Queryable
+
+Using the [#[derive(Queryable)]](http://docs.diesel.rs/diesel/query_source/trait.Queryable.html) trait annotation on your model struct allows records to be queried from the database. This means that any time you are using the `load()`, `get_result()`, `get_results()`, and `first()` methods to execute your queries, you must have `Queryable` implemented on your model.
+
+There are some considerations to be made when implementing the `Queryable` trait. You must include all the table's fields in your model struct. The implementation details for this rule are only important if you wish to `impl` the trait by hand and will be covered later in this section.  Your struct name should be the singular of its corresponding table name. Queryable by itself does not support the `#[table_name=""]` attribute. 
+
+### Example 
+
+*Note: src/lib.rs with #[macro_use] extern crate diesel_codegen; not shown*
+
+*src/models.rs*
+```rust
+#[derive(Queryable)]
+pub struct User {
+    pub id: i32,
+    pub first_name: String,
+    pub last_name: String,
+    pub email: String,
+}
+```
+
+*src/main.rs*
+```rust
+// externs, database connection, and using statement code omitted
+
+fn main() {
+    // The following will return all Users as a QueryResult<Vec<User>>
```

Comment:
[Historically](https://www.reddit.com/r/rust/comments/3fimgp/why_double_colon_rather_that_dot/ctozkd0/), people seem to like the name 'turbofish' more than its look. I personally prefer variable type annotations more than putting a turbofish at the end of a long chain of method calls, because you put the name right next to the type. But others may disagree, as the turbofish version of something like `.load::<Post>(&c)` or `.parse::<i32>()` can be read quite nicely as "load posts from connections" or "parse this as an integer".

You know what? I'll open an issue on the [fmt-rfcs](https://github.com/rust-lang-nursery/fmt-rfcs) repo.